src/acl/RegexData.cc

```diff
@@ -256,7 +251,9 @@ ACLRegexData::parse()
         }
     }
 
-    if (!compileOptimisedREs(data, sl)) {
+    auto options = dynamic_cast<Acl::CaseLineOptions *>(dirtyLineOptions());
+    assert(options);
+    if (!compileOptimisedREs(data, sl, options->isCaseInsensitive())) {
         debugs(28, DBG_IMPORTANT, "WARNING: optimisation of regular expressions failed; using fallback method without optimisation");
         compileUnoptimisedREs(data, sl);
```

Comment:
If compileOptimisedREs() is called with `options`, should not compileUnoptimisedREs() be called with `options`?

And, if yes, then perhaps we should replace the boolean "starting with a case insensitive" parameter with a "starting flags" parameter that will be computed once for both calls, before `sl` is computed?